app/controllers/category_recommendations_controller.rb

```diff
@@ -0,0 +1,58 @@
+class CategoryRecommendationsController < ApplicationController
+  include CustomControllerHelpers
+
+  before_action :authenticate_user
+  before_action :validate_namespace
+
+  def index
+    render_jsonapi recommendations
+  end
+
+  def realtime
+    render_jsonapi realtime_recommendations
+  end
+
+  private
+
+  def user
+    doorkeeper_token&.resource_owner
+  end
+
+  def recommendations_service
+    RecommendationsService::Media.new(user)
+  end
+
+  def recommendations
+    recommendations_service.category_recommendations_for(namespace_class)
+  end
+
+  def realtime_recommendations
+    recommendations_service.realtime_category_recommendations_for(
+      namespace_class
+    )
+  end
+
+  def namespace
+    @namespace ||= params[:namespace].classify
+  end
+
+  def namespace_class
+    @namespace_class ||= namespace.safe_constantize
+  end
+
+  def resource_class
+    "#{namespace}Resource".safe_constantize
+  end
+
+  def authenticate_user
```

Comment:
This might make sense in `CustomControllersHelper`, and maybe named `authenticate_user!` to match Devise (since the method has the same name)?